File: app/src/main/java/dev/ragnarok/fenrir/model/database/SchoolClazz.java
package dev.ragnarok.fenrir.model.database;

import android.os.Parcel;
import android.os.Parcelable;

public final class SchoolClazz implements Parcelable {

    public static final Creator<SchoolClazz> CREATOR = new Creator<SchoolClazz>() {
        @Override
        public SchoolClazz createFromParcel(Parcel in) {
            return new SchoolClazz(in);
        }

        @Override
        public SchoolClazz[] newArray(int size) {
            return new SchoolClazz[size];
        }
    };
    private final int id;
    private final String title;

    public SchoolClazz(int id, String title) {
        this.id = id;
        this.title = title;
    }

    private SchoolClazz(Parcel in) {
        id = in.readInt();
        title = in.readString();
    }

    public String getTitle() {
        return title;
    }

    public int getId() {
        return id;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(id);
        parcel.writeString(title);
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/model/database/SchoolClazz.kt
package dev.ragnarok.fenrir.model.database

import android.os.Parcel
import android.os.Parcelable

class SchoolClazz : Parcelable {
    val id: Int
    val title: String?

    constructor(id: Int, title: String?) {
        this.id = id
        this.title = title
    }

    internal constructor(`in`: Parcel) {
        id = `in`.readInt()
        title = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(id)
        parcel.writeString(title)
    }

    companion object CREATOR : Parcelable.Creator<SchoolClazz> {
        override fun createFromParcel(parcel: Parcel): SchoolClazz {
            return SchoolClazz(parcel)
        }

        override fun newArray(size: Int): Array<SchoolClazz?> {
            return arrayOfNulls(size)
        }
    }
}